plugins/inputs/syslog/nontransparent_test.go

```diff
@@ -269,33 +268,25 @@ func TestNonTransparentStrictWithZeroKeepAlive_tcp_tls(t *testing.T) {
 }
 
 func TestNonTransparentStrict_unix(t *testing.T) {
-	tmpdir, err := os.MkdirTemp("", "telegraf")
-	require.NoError(t, err)
-	defer os.RemoveAll(tmpdir)
-	sock := filepath.Join(tmpdir, "syslog.TestStrict_unix.sock")
+	tmpdir := t.TempDir()
+	sock := filepath.Join(tmpdir, "sock")
 	testStrictNonTransparent(t, "unix", sock, false, nil)
 }
 
 func TestNonTransparentBestEffort_unix(t *testing.T) {
-	tmpdir, err := os.MkdirTemp("", "telegraf")
-	require.NoError(t, err)
-	defer os.RemoveAll(tmpdir)
-	sock := filepath.Join(tmpdir, "syslog.TestBestEffort_unix.sock")
+	tmpdir := t.TempDir()
+	sock := filepath.Join(tmpdir, "sock")
 	testBestEffortNonTransparent(t, "unix", sock, false)
 }
 
 func TestNonTransparentStrict_unix_tls(t *testing.T) {
-	tmpdir, err := os.MkdirTemp("", "telegraf")
-	require.NoError(t, err)
-	defer os.RemoveAll(tmpdir)
-	sock := filepath.Join(tmpdir, "syslog.TestStrict_unix_tls.sock")
+	tmpdir := t.TempDir()
+	sock := filepath.Join(tmpdir, "sock")
 	testStrictNonTransparent(t, "unix", sock, true, nil)
 }
 
 func TestNonTransparentBestEffort_unix_tls(t *testing.T) {
-	tmpdir, err := os.MkdirTemp("", "telegraf")
-	require.NoError(t, err)
-	defer os.RemoveAll(tmpdir)
-	sock := filepath.Join(tmpdir, "syslog.TestBestEffort_unix_tls.sock")
+	tmpdir := t.TempDir()
+	sock := filepath.Join(tmpdir, "s")
```

Comment:
s instead of sock?

Maybe getting a socket should be in a separate function? Even though it's only two lines, they're repeated 10 times.

Reply:
See 1d673c7d7e68a1251daf467cdc42a419e8be515b. This is because maximum path length of Unix domain socket on MacOS is 104 characters.

You can see the failing tests here: https://app.circleci.com/pipelines/github/influxdata/telegraf/10063/workflows/a07abed0-b9a4-4397-9f5c-9abb6789b0dc/jobs/166791/tests#failed-test-16

Reply:
Refactored them into a function in `testutil` package. fa278d26eb0ea1877c556932f9eedadcbda887c5